snapcraft/internal/sources.py

```diff
@@ -119,8 +125,9 @@ def download(self):
 class Script(FileBase):
 
     def __init__(self, source, source_dir, source_tag=None,
-                 source_branch=None):
-        super().__init__(source, source_dir, source_tag, source_branch)
+                 source_branch=None, source_depth=None):
+        super().__init__(source, source_dir, source_tag,
+                         source_branch, source_depth)
```

Comment:
Can we call it 'git-source-depth'? Also items 19-21 in Effective Python might give you some ideas for simplifying the base class `__init__()` by popping the accepted kwargs and throwing an exception if there are any leftover options